pkg/arvo/app/group-hook.hoon

```diff
@@ -0,0 +1,270 @@
+/-  *group-store, *group-hook
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff [%group-update group-update]]
+      [%quit ~]
+      [%poke wire dock poke]
+      [%pull wire dock ~]
+      [%peer wire dock path]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  state-zero
+  $:  synced=(map path ship)
+      boned=(map wire (list bone))
+  ==
+::
++$  poke
+  $%  [%group-action group-action]
+      [%group-hook-action group-hook-action]
+  ==
+::
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    [~ this]
+  [~ this(+<+ u.old)]
+::
+++  poke-group-hook-action
+  |=  act=group-hook-action
+  ^-  (quip move _this)
+  ?-  -.act
+      %add
+    ?.  =(src.bol our.bol)
+      [~ this]
+    =/  group-path  [%group path.act]
+    =/  group-wire  [(scot %p ship.act) group-path]
+    ?:  (~(has by synced) path.act)
+      [~ this]
+    =.  synced  (~(put by synced) path.act ship.act)
+    :_  (track-bone group-wire)
+    ?:  =(ship.act our.bol)
+      [ost.bol %peer group-wire [ship.act %group-store] group-path]~
+    [ost.bol %peer group-wire [ship.act %group-hook] group-path]~
+  ::
+      %remove
+    =/  ship  (~(get by synced) path.act)
+    ?~  ship
+      [~ this]
+    ?:  &(=(u.ship our.bol) =(our.bol src.bol))
+      ::  delete one of our own paths
+      =/  group-wire  [(scot %p our.bol) %group path.act]
+      :_  this(synced (~(del by synced) path.act))
+      %+  weld
+        (pull-wire group-wire path.act)
+      ^-  (list move)
+      %+  turn  (prey:pubsub:userlib [%group path.act] bol)
+      |=  [=bone *]
+      ^-  move
+      [bone %quit ~]
+    ?:  |(=(u.ship src.bol) =(our.bol src.bol))
+      ::  delete a foreign ship's path
+      =/  group-wire  [(scot %p u.ship) %group path.act]
+      :_  this(synced (~(del by synced) path.act))
+      (pull-wire group-wire path.act)
+    :: don't allow
+    [~ this]
+  ::
+  ==
+::
+++  peer-group
```

Comment:
Am I correct in understanding that this exists solely so that foreign group-hook apps can peer this because we don't allow foreign apps to peer group-store?

Reply:
Yes.